test/gerkin/tests/features/data_delete.feature

```diff
@@ -0,0 +1,19 @@
+Feature: Remove record from the database.
+  Request endpoint: DELETE /data/{code}/{version}/delete
+
+  Background:
+    Given database with the valid schema has been set up
+
+  Scenario: Successfully delete existing record
+    When I make a DELETE request with a valid payload ID <ID>
```

Comment:
ID is not part of a payload but rather path param in URL, please change this. 
Generally I think request body in DELETE request is allowed, but I don't think it's a common approach and usually deleted entity is determined in path (as in API spec for this BB).